new coalignment api (first draft) (#207)
sunpy/sunkit-image: docs/how_to_guide/using_the_coalignment_interface.rst

Reviewer: Delete this.

Reviewer 2: I think all of the information here is already covered in the example gallery entry so I would recommend we remove this.

.. _sunkit-image-how-to-guide-using-the-coalignment-interface:

*******************************
Using the Coalignment Interface
*******************************

This guide explains how to use the coalignment interface to improve the alignment of solar images.

> The :func:`~sunkit_image.coalignment.coalignment` function facilitates image coalignment using a range of  registered methods.

Reviewer: ```suggestion
The :func:`~sunkit_image.coalignment.coalignment` function facilitates image coalignment using a range of registered methods.
```

Refer to the docstring of :func:`~sunkit_image.coalignment.coalignment` for detailed information on the parameters, return values, and exceptions.

Here is an example of how to use the :func:`~sunkit_image.coalignment.coalignment` function:

> .. code-block:: python

Reviewer: This should be a mpl plot code block instead.

    target_map = Map(sunpy.data.sample.AIA_193_CUTOUT03_IMAGE)
    ## Should match the platescale before coalignment so as to maintain correctness;
    nx= (reference_map.scale.axis1 * reference_map.dimensions.x )/target_map.scale.axis1
    ny= (reference_map.scale.axis2 * reference_map.dimensions.y )/target_map.scale.axis2
    aia_193_downsampled_map = reference_map.resample(u.Quantity([nx,ny]))
    # Coalign the target map to the reference map
    coaligned_map = coalignment(aia_193_downsampled_map, target_map, method="match_template")
    # Define contour levels
    levels = np.linspace(200, 1200, 5) * target_map.unit
    # Plotting
    fig = plt.figure(figsize=(15, 7.5))
    ax = fig.add_subplot(121, projection=target_map)
    target_map.plot(axes=ax, title='Before coalignment')
    bounds = ax.axis()
    aia_193_downsampled_map.draw_contours(levels, axes=ax, cmap='viridis', alpha=0.7)
    ax.axis(bounds)
    ax = fig.add_subplot(122, projection=coaligned_map)
    coaligned_map.plot(axes=ax, title='After coalignment')
    bounds = ax.axis()
    aia_193_downsampled_map.draw_contours(levels, axes=ax, cmap='viridis', alpha=0.7)
    ax.axis(bounds)

    plt.show()

> There is another example :ref:`sphx_glr_generated_gallery_adding_a_coalignment_method.py` focused on aligning an IS raster with an AIA image.

Reviewer 2: ```suggestion
There is another example :ref:`sphx_glr_generated_gallery_adding_a_coalignment_method.py` focused on aligning an EIS raster with an AIA image.
```

> 
> There is another example :ref:`sphx_glr_generated_gallery_adding_a_coalignment_method.py` focused on aligning an EIS raster with an AIA image.

Reviewer: ```suggestion
There is another example :ref:`sphx_glr_generated_gallery_adding_a_coalignment_method.py` focused on aligning an EIS raster with an AIA image.
```

> .. _sunkit-image-how-to-guide-using-the-coalignment-interface:

Reviewer: ```suggestion
.. _sunkit-image-how-to-guide-use-coalign:
```

*******************************
Using the Coalignment Interface
*******************************

> This guide explains how to use the coalignment interface to improve the alignment of solar images.
> The :func:`~sunkit_image.coalignment.coalign` function facilitates image coalignment using a range of registered methods.
> Refer to the docstring of :func:`~sunkit_image.coalignment.coalign` for detailed information on the parameters, return values, and exceptions.
> 
> Here is an example of how to use the :func:`~sunkit_image.coalignment.coalign` function:

Reviewer: ```suggestion
This guide explains how to use the :func:`~sunkit_image.coalignment.coalign` to improve the alignment of solar images.

Within the documentation string, there is detailed information on the parameters, return values, and exceptions.

For the code below, we will explain each step of the process:
```

    target_map = Map(sunpy.data.sample.AIA_193_CUTOUT03_IMAGE)
    ## Should match the platescale before coalignment so as to maintain correctness;
    nx= (reference_map.scale.axis1 * reference_map.dimensions.x )/target_map.scale.axis1
    ny= (reference_map.scale.axis2 * reference_map.dimensions.y )/target_map.scale.axis2
    aia_193_downsampled_map = reference_map.resample(u.Quantity([nx,ny]))
    # Coalign the target map to the reference map
    coaligned_map = coalign(aia_193_downsampled_map, target_map, method="match_template")
    # Define contour levels
    levels = np.linspace(200, 1200, 5) * target_map.unit
    # Plotting
    fig = plt.figure(figsize=(15, 7.5))
    ax = fig.add_subplot(121, projection=target_map)
    target_map.plot(axes=ax, title='Before coalignment')
    bounds = ax.axis()
    aia_193_downsampled_map.draw_contours(levels, axes=ax, cmap='viridis', alpha=0.7)
    ax.axis(bounds)
    ax = fig.add_subplot(122, projection=coaligned_map)
    coaligned_map.plot(axes=ax, title='After coalignment')
    bounds = ax.axis()
    aia_193_downsampled_map.draw_contours(levels, axes=ax, cmap='viridis', alpha=0.7)
    ax.axis(bounds)

    plt.show()

> There is another example :ref:`sphx_glr_generated_gallery_aligning_aia_with_eis_maps.py` focused on aligning an EIS raster with an AIA image.

Reviewer: ```suggestion
Within the Example Gallery, there is a simple example focused on aligning an EIS raster with an AIA image (:ref:`sphx_glr_generated_gallery_aligning_aia_with_eis_maps.py`).
```
